spec/models/setting_spec.rb

```diff
@@ -169,4 +169,40 @@
     end
   end
 
+  describe ".clear_cache!" do
+    before do
+      # prime the cache
+      Setting.host_name = 'foo'
+    end
+
+    it "doesn't fetch the setting from the database again" do
+      expect(Setting).to receive(:find_by_name).with('host_name').once
+
+      Setting.host_name
+      Setting.host_name
+    end
+
+    it "fetches the setting again when the cache was cleared" do
+      expect(Setting).to receive(:find_by_name).with('host_name').twice
+
+      Setting.host_name
+      Setting.clear_cache!
+      Setting.host_name
+    end
+
+    it "it doesn't affected any other cached values" do
+      Rails.cache.write('foo', 'bar')
+      Setting.clear_cache!
+      expect(Rails.cache.read('foo')).to eq 'bar'
+    end
+
+    it "falls back to clearing everything if partial deletion is unsupported" do
```

Comment:
switching between single and double-quotes based on the existence of a single-quote in the description itself is just wrong, imo. look at the test above which contains `'`.